tests/test_jenkins.py

```diff
@@ -0,0 +1,292 @@
+        self.assertEqual(len(cached_builds), len(builds))
+
+        with open("data/jenkins_build.json") as build_json:
+            first_build = json.load(build_json)
+            self.assertDictEqual(cached_builds[0]['data'], first_build['data'])
+
+
+    def test_fetch_from_empty_cache(self):
+        """Test if there are not any builds returned when the cache is empty"""
+
+        cache = Cache(self.tmp_path)
+        jenkins = Jenkins(JENKINS_SERVER_URL, job_url=JENKINS_JOB_BUILDS_URL,  cache=cache)
+        cached_builds = [build for build in jenkins.fetch_from_cache()]
+        self.assertEqual(len(cached_builds), 0)
+
+    def test_fetch_from_non_set_cache(self):
+        """Test if a error is raised when the cache was not set"""
+
+        jenkins = Jenkins(JENKINS_SERVER_URL, job_url=JENKINS_JOB_BUILDS_URL)
+
+        with self.assertRaises(CacheError):
+            _ = [build for build in jenkins.fetch_from_cache()]
+
+
+class TestJenkinsCommand(unittest.TestCase):
+
+    @httpretty.activate
+    def test_parsing_on_init(self):
+        """Test if the class is initialized"""
+
+        args = ['--job-url', JENKINS_JOB_BUILDS_URL,
+                '--origin', 'test',
+                '--site', JENKINS_SERVER_URL]
+
+        cmd = JenkinsCommand(*args)
+        self.assertIsInstance(cmd.parsed_args, argparse.Namespace)
+        self.assertEqual(cmd.parsed_args.job_url, JENKINS_JOB_BUILDS_URL)
+        self.assertEqual(cmd.parsed_args.site, JENKINS_SERVER_URL)
+        self.assertEqual(cmd.parsed_args.origin, 'test')
+        self.assertIsInstance(cmd.backend, Jenkins)
+
+    def test_argument_parser(self):
+        """Test if it returns a argument parser object"""
+
+        parser = JenkinsCommand.create_argument_parser()
+        self.assertIsInstance(parser, argparse.ArgumentParser)
+
+
+class TestJenkinsClient(unittest.TestCase):
+    """Jenkins API client tests
+
+    These tests not check the body of the response, only if the call
+    was well formed and if a response was obtained. Due to this, take
+    into account that the body returned on each request might not
+    match with the parameters from the request.
+    """
+    @httpretty.activate
+    def test_init(self):
+        """Test initialization"""
+        client = JenkinsClient(JENKINS_SERVER_URL)
+
+    @httpretty.activate
+    def test_get_jobs(self):
+        """Test get_jobs API call"""
+
+        # Set up a mock HTTP server
+        body = read_file('data/jenkins_jobs.json')
+        httpretty.register_uri(httpretty.GET,
+                               JENKINS_JOBS_URL,
+                               body=body, status=200)
+
+        client = JenkinsClient(JENKINS_SERVER_URL)
+        response = client.get_jobs()
+
+        self.assertEqual(response, body)
+
+    @httpretty.activate
+    def test_get_builds(self):
+        """Test get_builds API call"""
+
+        # Set up a mock HTTP server
+        body = read_file('data/jenkins_jobs.json')
```

Comment:
Hmmm, this is the list of jobs that have the builds. Then you have the builds for each job in a separate file.